### DISCUSSION TITLE
[listen on textbox fill color changed ]

### DISCUSSION THREAD

**Discussion by Othman2001** [fabricjs/fabric.js#7890] [Q&A] [answered]
i am working on building full design editor using fabric js, and i am updating react-design-editor  https://salgum1114.github.io/react-design-editor/
but i am facing problem, i am build. a min preview where any change happens on the canvas the user will be able to see these changes on min preview 
you can have look on this screeshot:
https://firebasestorage.googleapis.com/v0/b/print-on-demand-7c2eb.appspot.com/o/Screen%20Shot%202022-04-19%20at%203.41.54%20PM.png?alt=media&token=d6dff2b2-181f-4476-9b85-47454c51820b

so to make this min preview i manged to make on ObjectModified:event and then convert my canvas to dataUrl and render this dataUrl on image 
`            this.canvasRef.canvas.on('object:modified', function () {
                updateDesign({
                    editorCanvas: { canvasObjectsJSON: canvas.toJSON() },
                });
                setDataUrl({ dataUrl: canvas.toDataURL() });
            });`

and for now everything works fine, but the problem when i change the fill color of textbox/ text it is not updated 
any ideas on how can i fix this issue

**Comment by ShaMan123** [comment, marked as answer]
Well those changes don't fire a modified event.
override the `_set` method on Object (then you can fire event from there... I do that) and make sure you modify objects using the `set` method instead of directly (`object.fill = 'red'` should be `object.set({fill:'red'})`)

http://fabricjs.com/events
http://fabricjs.com/build-minimap

**Reply by asturur** [reply]
Yes in general, if you write code to make a change happen, you don't need an event.
you can either:

- write your own function that make the change and execute the extra code
- fire the event on your own, understanding that you go in extra loops in order to execute some code
- or override the _set method

The simplest and safest solution is to write your own function

function mySetValues(object, property, value) {
  object.set(property, value);
 ... all the rest of my businnes logic
}

if this needs to happen also for fabricJS internals, you should override the _set method or listen to all existing events in addition to your custom function.

rewriting the _set method puts you a bit on the hook with fabricJS updates, i have been there too with one application and while is doable, is an extra effort

**Reply by ShaMan123** [reply]
Same here
